src/views/Home.jsx

```diff
@@ -35,7 +44,26 @@ export function Home({ setListToken }) {
 				navigate('/list');
 			})
 			.catch((error) => {
-				errorNotify();
+				listCreateError();
+			});
+	};
+
+	// Join existing shopping list
+	const joinExistingToken = async (e) => {
+		e.preventDefault();
+
+		queryForCollection(joinToken)
+			.then((exists) => {
+				if (exists) {
+					setListToken(joinToken);
+					navigate('/list');
+				} else {
+					setJoinToken('');
+					listJoinError();
+				}
+			})
+			.catch((error) => {
+				blankError();
```

Comment:
I've spent a while trying to wrap my head around this and the `queryForCollection()` function from firebase.js, and I'm just super impressed with how you got here! Just to make sure I understand correctly--is the `exists` in the `.then` statement basically referring to if `queryForCollection(joinToken)` returns true? Great work on this!

Reply:
@sarahmjenkins Indeed, you’re right! `.then` is expecting a boolean. Though since `exists` is a parameter it could technically be any word (like `fluffybunnies`! 🐰)

Reply:
I was going to ask what condition leads `queryForCollection` to throw an error, but I'm realizing it's when you check for existence with a blank list token. I made a comment somewhere else about validating the user input.

Reply:
I was also wondering about the `.then`.  Is this checking against something specific in firebase?

Reply:
great job figuring this out!